### ISSUE TITLE
check_disk : --freespace-ignore-reserved should be the default

### ISSUE THREAD

**Issue by yhuelf** [monitoring-plugins/monitoring-plugins#1816]
Hello,

It seems to me that the option `--freespace-ignore-reserved` should be the default.

It would be consistent with the computation of the percentage : 

```
  p->dused_pct = calculate_percent( p->used, p->used + p->available );    /* used + available can never be > uintmax */
  p->dfree_pct = 100 - p->dused_pct;
```

In the current situation, we can have a WARNING (or CRITICAL) alert, while the displayed values in the perfdata show that usage is less than threshold. For instance :

```
user@host:~$ /usr/local/libexec/check_disk -w 82% -c 10% -m -p /luks_headers_backup/ --freespace-ignore-reserved
DISK WARNING - free space: /luks_headers_backup 142MiB (81% inode=99%);| /luks_headers_backup=33554432B;32841400;164207001;0;182452224

user@host:~$ /usr/local/libexec/check_disk -w 82% -c 10% -m -p /luks_headers_backup/
DISK WARNING - free space: /luks_headers_backup 142MiB (81% inode=99%);| /luks_headers_backup=33554432B;35483811;177419059;0;197132288

root@host:~# tune2fs -l /dev/sda4 | grep -i 'reserved block count\|block size'
Reserved block count:     10240
Block size:               1024
```

**Comment by RincewindsHat** [comment]
Hi, sorry for not answering for so long.
I am not sure if I understand the situation fully, but I think there is some thought behind this option not being the default.
My reasoning here would be the following one: Most (useful) operations on a system  are done by non-root users (webserver, database, etc), so it is kinda important whether they are able to write data to the disk.
I we choose to add root reserved blocks to the total by default, you could run into the situation, where a process would be unable to write to the filesystem anymore (all unreserved blocks are used), but the monitoring would not warn an administrator about it.

Now I am wondering whether the computation of the percentage is correct in that sense though.

**Comment by yhuelf** [comment]
Hi, thanks for your answer.

Your reasoning is most probably right. Actually, it doesn't really matter to me wich is the default, as long as the numbers are consistent. I think that the computation of the percentage should be fixed and take into account the option `--freespace-ignore-reserved` (wether it is set or not).

**Comment by RincewindsHat** [comment]
@yhuelf  Indeed, you are right about the inconsistency. I will take a look, if I find the time.
